doc/variants.rst

```diff
@@ -0,0 +1,133 @@
+****************************
+Virtual Libraries & Variants
+****************************
+
+Virtual libraries correspond to dune's ability to compile parameterized
+libraries and delay the selection of concrete implementations until linking an
+executable.
+
+The feature introduces two kinds of libraries: virtual and implementations. A
+*virtual library* corresponds to an interface (although it may contain partial
+implementation). An *implementation* of a virtual library fills in all
+unimplemented modules in the virtual library.
+
+The benefit of this partition is that other libraries may depend and compile
+against the virtual library and only select concrete implementations for these
+virtual libraries when linking executables. An example where this might be
+useful would be a virtual, cross platform, ``os`` library. This library would
```

Comment:
What do you think about making this example a bit more concrete? For example:
- a "clock" virtual library that has two implementations, "clock.unix" and "clock.win",
- a "calendar" library depending on "clock" that deals with time & date calculations,
- a "birthday-reminder" executable depending on "calendar" that computes your friends' birthdays.

Reply:
Sure. But do you mean with actual code? Or just change it to describe your example?

Reply:
I was just thinking about changing the names to describe the example. Some example code under `example/` would be nice one day, but we'll probably need something simpler.

Reply:
Ah ok. Fixed the examples to use clock.